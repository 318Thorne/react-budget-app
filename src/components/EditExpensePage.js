import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {editExpense, startRemoveExpense} from '../actions/expenses';


export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
            this.props.onSubmit(this.props.expense.id, expense);
            this.props.history.push('/');
        };
    onClick = () => {
            const id = this.props.expense.id
            this.props.onClick(id);
            this.props.history.push('/');
        };

    render(){
        return (
            <div>
                <ExpenseForm 
                    expense = {this.props.expense}
                    onSubmit={this.onSubmit}                  
                />
                <button onClick={this.onClick}>Remove item</button>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        onSubmit: (id, expense) => dispatch(editExpense(id, expense)),
        onClick: (id) => dispatch(startRemoveExpense({id}))       
    }
}

const mapStateToProps = (state, props) => {
    return{
        expense: state.expenses.find((expense) => (expense.id === props.match.params.id) )
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);