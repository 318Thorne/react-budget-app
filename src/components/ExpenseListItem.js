import React from 'react';
import {Link} from 'react-router-dom';


export const ExpenseListItem = ({id , description, amount, createdAt}) => (
    <div>
        <p>{description}</p>
        <p>{amount}</p>
        <p>{createdAt}</p>
        <Link to={`/edit/${id}`}>edit</Link>
        <hr/>
    </div>
);

// const mapStateToProps = (state) => {
//     return {
//         expenses: state.expenses
//     }
// }

export default ExpenseListItem