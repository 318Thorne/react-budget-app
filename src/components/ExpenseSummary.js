import React from 'react';
import { connect } from 'react-redux';
import ExpenseTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral'

export const ExpenseSummary = ({expensesCount, expensesTotal}) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses'
    return (
        <div>            
            <p>Viewing {expensesCount} {expenseWord} totalling {numeral(expensesTotal / 100).format('$0,0.00')} </p>
        </div>
    )
}

const mapStateToProps = (state) => {
    const expenses = selectExpenses(state.expenses, state.filters)
    return {
        expensesCount: expenses.length,
        expensesTotal: ExpenseTotal(expenses)
    }
}

export default connect(mapStateToProps)(ExpenseSummary);