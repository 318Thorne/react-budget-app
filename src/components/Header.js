import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <ul>
            <li><NavLink to="/" activeClassName="isActive" exact={true}>Home</NavLink></li>
            <li><NavLink to="/create" activeClassName="isActive">Add</NavLink></li>
            <li><NavLink to="/help" activeClassName="isActive" >Help</NavLink></li>
        </ul>
    </header>
);
export default Header;