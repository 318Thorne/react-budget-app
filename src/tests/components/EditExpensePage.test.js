import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses';

let wrapper, onSubmit, onClick, history;

beforeEach(() => {
    history = { push: jest.fn() };
    onSubmit = jest.fn();
    onClick = jest.fn();
    wrapper = shallow(<EditExpensePage expense={expenses[1]} onSubmit={onSubmit} onClick={onClick} history={history}/>);
})

test('should render edit expense page correctly', () => {
    expect(wrapper).toMatchSnapshot();
})

test('should call onSubmit', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
    expect(history.push).toHaveBeenCalledWith('/');
    expect(onSubmit).toHaveBeenCalledWith(expenses[1].id,expenses[1]);
})

test('should call onClick', () => {
    wrapper.find('button').prop('onClick')(expenses[1]);
    expect(history.push).toHaveBeenCalledWith('/');
    expect(onClick).toHaveBeenCalledWith(expenses[1].id);
})