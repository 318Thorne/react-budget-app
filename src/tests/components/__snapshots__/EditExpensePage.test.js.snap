// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render edit expense page correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditExpensePage
    expense={
      Object {
        "amount": 197000,
        "createdAt": -345600000,
        "description": "rent",
        "id": "2",
        "note": "",
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    onClick={[MockFunction]}
    onSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ExpenseForm
          expense={
            Object {
              "amount": 197000,
              "createdAt": -345600000,
              "description": "rent",
              "id": "2",
              "note": "",
            }
          }
          onSubmit={[Function]}
        />,
        <button
          onClick={[Function]}
        >
          Remove item
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "expense": Object {
            "amount": 197000,
            "createdAt": -345600000,
            "description": "rent",
            "id": "2",
            "note": "",
          },
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Remove item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Remove item",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ExpenseForm
            expense={
              Object {
                "amount": 197000,
                "createdAt": -345600000,
                "description": "rent",
                "id": "2",
                "note": "",
              }
            }
            onSubmit={[Function]}
          />,
          <button
            onClick={[Function]}
          >
            Remove item
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "expense": Object {
              "amount": 197000,
              "createdAt": -345600000,
              "description": "rent",
              "id": "2",
              "note": "",
            },
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Remove item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Remove item",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
